packages/hyperion-autologging/src/ALUIEventPublisher.ts

```diff
@@ -314,6 +314,7 @@ export function publish(options: InitOptions): void {
         surface,
         surfaceData,
         ...elementText,
+        metadata: {...uiEventData.metadata,...surfaceData?.metadata},
```

Comment:
Wondering what should take precedence here in this case if there is a key conflict,  should we reverse this where ui metadata takes precedence over the surface metadata?